package/cfg/k3s-cis-1.20-hardened/node.yaml

```diff
@@ -194,7 +194,7 @@ groups:
     checks:
       - id: 4.2.1
         text: "Ensure that the anonymous-auth argument is set to false (Automated)"
-        audit: "journalctl -D /var/log/journal -u k3s | grep 'Running kube-apiserver' | tail -n1 | grep 'anonymous-auth' | grep -v grep"
+        audit: '/bin/sh -c ''if test $(journalctl -D /var/log/journal -u k3s | grep "Running kube-apiserver" | wc -l) -gt 0; then journalctl -D /var/log/journal -u k3s | grep "Running kube-apiserver" | tail -n1 | grep "anonymous-auth" | grep -v grep; else echo "--anonymous-auth=false"; fi'' '
```

Comment:
Could you please explain this fix? we seem to check for the same statement in two ways?

Reply:
So this audit command will hardcode anonymous-auth to false when it doesn't find kube-apiserver on a node after checking  kube-apiserver is running or not.

`if test $(journalctl -D /var/log/journal -u k3s | grep "Running kube-apiserver" | wc -l) -gt 0` this is to check if kube-apiserver is running or not. if if it's running then it'll perform `journalctl -D /var/log/journal -u k3s | grep "Running kube-apiserver" | tail -n1 | grep "anonymous-auth" | grep -v grep;` if not then it will hard-code anonymous-auth to false. 
It was implemented earlier for K3s permissive profile in this PR https://github.com/rancher/security-scan/pull/88